src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -93,6 +155,7 @@ protected override void Dispose(bool disposing)
             state.CancellationToken = cancellationToken;
             state.RequestMessage = request;
             state.Handler = this;
+            state.ProxyPolicy = _proxyPolicy;
```

Comment:
This code below is wrong:
>
            if (value)
            {
                _proxyPolicy = CurlProxyUsePolicy.UseCustomProxy;
            }

You should set the policy to .UseCustomProxy.  That is ONLY true if the Proxy property is non-NULL.  Otherwise, setting UseProxy to true but having a NULL Proxy property should mean to use the "default proxy" which in your case means check the environment variable.

Because you can't have order dependencies in assuming the setting of the .Proxy property before the .UseProxy property, you need to track the .UseProxy separately as a boolean internally.  Then you need to resolve the logic of combinations of .UseProxy and .Proxy in your .SendAsync() method.

Please look at the HttpClientHandler.Windows.cs file.